pkg/envoy/ads/cache_stream.go

```diff
@@ -15,15 +15,17 @@ import (
 	"github.com/openservicemesh/osm/pkg/constants"
 	"github.com/openservicemesh/osm/pkg/envoy"
 	"github.com/openservicemesh/osm/pkg/errcode"
-	"github.com/openservicemesh/osm/pkg/k8s/events"
 )
 
 // Routine which fulfills listening to proxy broadcasts
 func (s *Server) broadcastListener() {
-	// Register to Envoy global broadcast updates
-	broadcastUpdate := events.Subscribe(announcements.ProxyBroadcast)
+	// Register for proxy config updates broadcasted by the message broker
+	proxyUpdatePubSub := s.msgBroker.GetProxyUpdatePubSub()
+	proxyUpdateChan := proxyUpdatePubSub.Sub(announcements.ProxyUpdate.String())
+	defer s.msgBroker.Unsub(proxyUpdatePubSub, proxyUpdateChan)
```

Comment:
This code is buggy even prior to this PR, because it does not accept a channel where it can be signalled to stop. The Unsub is syntactically correct here, and clearly implies that it must happen should the function return, which will be the case when we update this function to accept a receive channel.
FYI, this is unused and is related to the snapshot cache feature that's not usable.

Reply:
this code is feature flagged and can be safely removed altogether if cache-based xds doesn't materialize going forward